tests/source/versionsort_non_ascii_numerics.rs

```diff
@@ -0,0 +1,33 @@
+use std::cmp::Ordering;
+use print๙msg::print as first_print;
+use print0msg::print as second_print;
+use printémsg::print as third_print;
```

Comment:
Added. The 2015 edition actually sorts the same way since `U+0030 < U+00E9 < U+0E59`, so the bug only affected 2024 edition (and can't be reproduced on earlier versions because ASCII digits are the earliest Unicode numerics).

Reply:
Thanks for adding that additional test case!